x/trigger/simulation/genesis.go

```diff
@@ -0,0 +1,140 @@
+package simulation
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"time"
+
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	simtypes "github.com/cosmos/cosmos-sdk/types/simulation"
+	sdktx "github.com/cosmos/cosmos-sdk/types/tx"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+
+	"github.com/provenance-io/provenance/internal/pioconfig"
+	"github.com/provenance-io/provenance/x/trigger/types"
+)
+
+const (
+	TriggerID  = "trigger_id"
+	QueueStart = "queue_start"
+)
+
+// TriggerIDFn randomized MinActions
+func TriggerIDFn(r *rand.Rand) uint64 {
+	return uint64(simtypes.RandIntBetween(r, 2, 10000000000))
+}
+
+// QueueStartFn randomized MinActions
+func QueueStartFn(r *rand.Rand) uint64 {
+	return uint64(simtypes.RandIntBetween(r, 0, 10000000000))
+}
+
+// GetRandomEvent returns a random event
+func GetRandomEvent(r *rand.Rand, now time.Time) types.TriggerEventI {
+	rand := simtypes.RandIntBetween(r, 0, 1)
+	if rand > 0 {
+		randTime := now.Add(time.Hour * time.Duration(simtypes.RandIntBetween(r, 1, 10)))
+		return &types.BlockTimeEvent{Time: randTime.UTC()}
+	}
+	height := uint64(simtypes.RandIntBetween(r, 100000, 200000))
+	return &types.BlockHeightEvent{BlockHeight: height}
+}
+
+// GetRandomAction returns a random action
+func GetRandomAction(r *rand.Rand, from string, to string) sdk.Msg {
+	amount := int64(simtypes.RandIntBetween(r, 100, 1000))
+	return &banktypes.MsgSend{
+		FromAddress: from,
+		ToAddress:   to,
+		Amount:      sdk.NewCoins(sdk.NewInt64Coin(pioconfig.GetProvenanceConfig().BondDenom, amount)),
+	}
+}
+
+// GetRandomTrigger returns a random trigger
+func GetRandomTrigger(r *rand.Rand, simState *module.SimulationState) types.Trigger {
+	account := int64(simtypes.RandIntBetween(r, 1, 2))
+	event := GetRandomEvent(r, simState.GenTimestamp.UTC())
+	action := GetRandomAction(r, simState.Accounts[0].Address.String(), simState.Accounts[account].Address.String())
+
+	actions, err := sdktx.SetMsgs([]sdk.Msg{action})
+	if err != nil {
+		panic("SetMsgs failed for GetRandomTrigger")
+	}
+	anyEvent, err := codectypes.NewAnyWithValue(event)
+	if err != nil {
+		panic("NewAnyWithValue failed for GetRandomTrigger")
+	}
+	return types.NewTrigger(1, simState.Accounts[0].Address.String(), anyEvent, actions)
+}
+
+// GetRandomQueuedTrigger returns a random queued trigger
+func GetRandomQueuedTrigger(_ *rand.Rand, simState *module.SimulationState) types.QueuedTrigger {
+	var event types.TriggerEventI = &types.BlockHeightEvent{BlockHeight: 100000}
```

Comment:
It'd be better to pick a random block height that has a chance of being reached. I think the longest sims do 500 blocks, but the longest that get regularly run only do 100. So maybe a random number between 10 and 150.

Reply:
I updated GetRandomTrigger with your height suggestion. I think this one should actually be between [0,currentHeight]. This is the height that they were actually added to the Queue.

Reply:
It's being added at genesis, but the hight that it's supposed to trigger at should be at least a few blocks out so that it actually gets triggered.

Also, we might not want to allow a height of 0 since that's never possible.